### PR TITLE
auth and global toasts

### PR DESCRIPTION
**Pull request by ahmadmoaaz2**
added login and added error and message handler for entire app

### CODE REVIEW

File: frontend/src/components/modal/ModalLoginView.js [hunk 1]
```diff
@@ -46,6 +77,9 @@ function ModalLoginView(props) {
                     }}
                     size="small"
                     variant="outlined"
+                    required
+                    defaultValue={'new@test.tes'}
```
Comment (Nikko0809) [inline frontend/src/components/modal/ModalLoginView.js:81]: remove defaultValue

File: frontend/src/components/modal/ModalLoginView.js [hunk 2]
```diff
@@ -55,6 +89,9 @@ function ModalLoginView(props) {
                     size="small"
                     autoComplete="current-password"
                     variant="outlined"
+                    required
+                    defaultValue={'Testtest1'}
```
Comment (Nikko0809) [inline frontend/src/components/modal/ModalLoginView.js:93]: remove defaultValue

File: frontend/src/components/nav/Navbar.js [hunk 3]
```diff
@@ -75,11 +77,18 @@ function Navbar() {
...
-    let [auth, setAuth] = useState(false);
+    let [auth, setAuth] = useState();
     let [anchor1, setAnchor1] = useState(null);
+    useEffect(() => {
+        setAuth(sessionStorage.getItem('firstName'));
+    }, [sessionStorage.getItem('token')])
     const handleMenu1 = (event) => setAnchor1(event.currentTarget);
-    // TODO: Remove setAuth
-    const handleClose1 = () => {setAnchor1(null);setAuth(false)}
+    const handleClose1 = () => setAnchor1(null);
```
Comment (Nikko0809) [inline frontend/src/components/nav/Navbar.js:86]: remove 1 from the variable

File: frontend/src/components/nav/Navbar.js [hunk 4]
```diff
@@ -112,7 +121,7 @@ function Navbar() {
                         handleClose1();
                         window.location.href = "/account-settings/orders"
                     }}>My account</MenuItem>
-                    <MenuItem onClick={handleClose1}>Logout</MenuItem>
```
Comment (Nikko0809) [inline frontend/src/components/nav/Navbar.js:115]: remove the 1 from handleClose

File: frontend/src/components/modal/ModalLoginView.js [hunk 5]
```diff
@@ -30,12 +32,41 @@ const useStyles = makeStyles((theme) => ({
...
+        });
+        let resJson = await response.json();
+        if (response.status === 200){
+            sessionStorage.setItem('firstName', resJson.firstName);
+            sessionStorage.setItem('lastName', resJson.lastName);
+            sessionStorage.setItem('email', resJson.email);
+            sessionStorage.setItem('token', resJson.token);
+            dispatch(setToast({type: "success", duration: 4000, alert: "Successfully Logged in... Redirecting...", onClose: () => window.location.href = "/"}))
+        } else if (response.status === 401){
+            dispatch(setToast({type: "warning", alert: resJson.message, duration: 10000}));
```
[the remark this answers is not in the document]
Comment (pavi102) [reply frontend/src/components/modal/ModalLoginView.js:59]: Yes change it to type error.  Also make sure the modal re-routes to the current page that the user is on and not the home page every time.

**Review by pavi102 — APPROVED** [review]

**Pull request merged**